src/ol/render/canvas/canvasreplay.js

```diff
@@ -343,8 +343,9 @@ ol.render.canvas.Replay.prototype.replay_ = function(
             context.globalAlpha = alpha * opacity;
           }
 
-          context.drawImage(image, originX, originY, width, height,
-              x, y, width * pixelRatio, height * pixelRatio);
+          context.drawImage(image, originX, originY, width - originX,
+              height - originY, x, y, (width - originX) * pixelRatio,
+              (height - originX) * pixelRatio);
```

Comment:
Very very minor: For readability it'd be nice if thes eypressions would be variables. No must, of course